### PR TITLE
Pod spec v1 changes

### PR DESCRIPTION
**Pull request by rgulewich**
The v1 pod spec (see [here](https://github.com/Netflix/titus-kube-common/blob/master/docs/examples/complete-pod.yaml)) changes the names of various annotations, so use them if on a v1 pod.

### CODE REVIEW

File: titus_isolate/kub/constants.py [hunk 1]
```diff
@@ -5,4 +5,11 @@
...
-UNKNOWN_INSTANCE_TYPE = 'unknown'
\ No newline at end of file
+UNKNOWN_INSTANCE_TYPE = 'unknown'
+
+# v1 pod spec constants
+ANNOTATION_KEY_POD_SPEC_VERSION = 'pod.netflix.com/pod-schema-version'
+V1_ANNOTATION_KEY_JOB_ID = 'v3.job.titus.netflix.com/id'
+V1_ANNOTATION_KEY_JOB_TYPE = 'v3.job.titus.netflix.com/type'
+V1_ANNOTATION_KEY_OWNER_EMAIL = 'workload.netflix.com/owner-email'
+V1_ANNOTATION_KEY_CPU_BURSTING = 'pod.netflix.com/cpu-bursting-enabled'
```
Comment (gabrielhartmann) [inline titus_isolate/kub/constants.py:15]: This feature is not a thing anymore.

File: titus_isolate/model/kubernetes_workload.py [hunk 2]
```diff
@@ -204,3 +194,43 @@ def __get_serializable_pod(self, pod: V1Pod) -> dict:
...
+def get_job_annotations_from_pod(pod: V1Pod) -> dict:
+    metadata = pod.metadata
+    job_id = None
+    workload_type = STATIC
+
+    if metadata.annotations.get(ANNOTATION_KEY_POD_SPEC_VERSION) is not None:
+        job_id = metadata.annotations.get(V1_ANNOTATION_KEY_JOB_ID, None)
+        job_type = metadata.annotations.get(V1_ANNOTATION_KEY_JOB_TYPE, None)
+        owner_email = metadata.annotations.get(V1_ANNOTATION_KEY_OWNER_EMAIL, None)
+        workload_type_str = metadata.annotations.get(V1_ANNOTATION_KEY_CPU_BURSTING, None)
```
Comment (gabrielhartmann) [inline titus_isolate/model/kubernetes_workload.py:207]: e.g. instead of looking up the annotation, we can just always return "STATIC"

**Review by gabrielhartmann — APPROVED** [review]

File: titus_isolate/model/kubernetes_workload.py (hunk 2 above)
Comment (rgulewich) [reply titus_isolate/model/kubernetes_workload.py:207]: OK, great - updated with that change.

Comment (gabrielhartmann) [reply titus_isolate/model/kubernetes_workload.py:207]: I still see references to `V1_ANNOTATION_KEY_CPU_BURSTING` annotation.  Maybe you forgot to push?

Comment (rgulewich) [reply titus_isolate/model/kubernetes_workload.py:207]: I explicitly added tests to make sure that even if you set `V1_ANNOTATION_KEY_CPU_BURSTING`, the type is set to `STATIC`.

Comment (gabrielhartmann) [reply titus_isolate/model/kubernetes_workload.py:207]: I don't understand. Maybe github is lying to me about the state of code, but I see the annotations being read here and
```
workload_type = BURST
```
being set sometimes.  I think we should just drop all reference to cpu bursting, as done here:
https://github.com/Netflix-Skunkworks/titus-isolate/pull/340/files

Comment (gabrielhartmann) [reply titus_isolate/model/kubernetes_workload.py:207]: It's fine.  If your changes work, we can merge them then merge the other PR.

File: titus_isolate/model/kubernetes_workload.py [hunk 3]
```diff
@@ -84,14 +74,14 @@ def __init_metadata(self, pod: V1Pod):
...
         self.__app_name = app_name
         self.__image = image
         self.__command = command
         self.__entrypoint = entrypoint
-        self.__job_type = job_type
-        self.__owner_email = owner_email
-        self.__workload_type = workload_type
+        self.__job_type = job_meta.get('job_type')
+        self.__owner_email = job_meta.get('owner_email')
+        self.__workload_type = job_meta.get('workload_type')
```
Comment (gabrielhartmann) [inline titus_isolate/model/kubernetes_workload.py:84]: We can just set
```
self.__workload_typoe = STATIC
```
here.

File: titus_isolate/model/kubernetes_workload.py [hunk 4]
```diff
@@ -204,3 +194,37 @@ def __get_serializable_pod(self, pod: V1Pod) -> dict:
 
 def get_workload_from_pod(pod: V1Pod) -> KubernetesWorkload:
     return KubernetesWorkload(pod)
+
+def get_job_annotations_from_pod(pod: V1Pod) -> dict:
+    metadata = pod.metadata
+    job_id = None
+    workload_type = STATIC
```
Comment (gabrielhartmann) [inline titus_isolate/model/kubernetes_workload.py:201]: this could be removed

File: titus_isolate/model/kubernetes_workload.py [hunk 5]
```diff
@@ -204,3 +194,37 @@ def __get_serializable_pod(self, pod: V1Pod) -> dict:
...
+    if metadata.annotations.get(ANNOTATION_KEY_POD_SPEC_VERSION) is not None:
+        job_id = metadata.annotations.get(V1_ANNOTATION_KEY_JOB_ID, None)
+        job_type = metadata.annotations.get(V1_ANNOTATION_KEY_JOB_TYPE, None)
+        owner_email = metadata.annotations.get(V1_ANNOTATION_KEY_OWNER_EMAIL, None)
+
+        return {
+            'job_id': job_id,
+            'job_type': job_type,
+            'owner_email': owner_email,
+            'workload_type': workload_type,
```
Comment (gabrielhartmann) [inline titus_isolate/model/kubernetes_workload.py:212]: this could be removed

File: titus_isolate/model/kubernetes_workload.py [hunk 6]
```diff
@@ -204,3 +194,37 @@ def __get_serializable_pod(self, pod: V1Pod) -> dict:
...
+        # legacy, very few pods launched a while ago
+        job_id = metadata.annotations.get(ANNOTATION_KEY_JOB_ID, 'UNKNOWN_JOB_ID')
+    job_type = metadata.annotations[WORKLOAD_JSON_JOB_TYPE_KEY]
+    owner_email = metadata.annotations[OWNER_EMAIL]
+    
+    return {
+        'job_id': job_id,
+        'job_type': job_type,
+        'owner_email': owner_email,
+        'workload_type': workload_type,
```
Comment (gabrielhartmann) [inline titus_isolate/model/kubernetes_workload.py:229]: this could be removed

**Review by gabrielhartmann — APPROVED** [review]

**Pull request merged**